src/autopr/SubModUpdate.d

```diff
@@ -0,0 +1,182 @@
+    string tree_sha;
+    /// Branch name used for the update
+    string branch;
+
+
+    /***************************************************************************
+
+        Creates or updates an existing PR for this update
+
+        Params:
+            con = octod connection reference to use
+            fork_owner = owner of the fork from which the PR will be created
+            fork_name = name the fork repository has
+
+        Returns:
+            Whether the PR was created or only updated (or no action was taken)
+
+    ***************************************************************************/
+
+    void createOrUpdatePR ( ref HTTPConnection con, string fork_owner,
+        string fork_name )
+    {
+        import autopr.github;
+        import std.stdio;
+        import std.format;
+
+        // Create Tree with updated submodule
+        auto tree = con.createTree(fork_owner,
+            fork_name, this.tree_sha,
+            format("submodules/%s", this.submod.name),
+            this.submod_new.sha, Type.Submodule);
+
+        writefln("%s> Created updated tree for %s (%s)",
+            this.repo, this.submod, tree["sha"].get!string);
+
+        auto commit_msg = format("Update %s to %s\n\n%s %s(%s)...%s(%s)",
+           this.submod.name, this.submod_new.ver, this.submod.name,
+           this.submod_cur.ver, this.submod_cur.sha[0..7],
+           this.submod_new.ver, this.submod_new.sha[0..7]);
+
+        auto commit = con.createCommit(fork_owner,
+            fork_name, this.commit_sha, tree["sha"].get!string, commit_msg);
+
+        writefln("%s> Created commit for %s (%s)",
+            this.repo, this.submod, commit["sha"].get!string);
+
+        import vibe.data.json;
+
+        Json reference;
+
+        auto refname = format("refs/heads/update-%s", this.submod.name);
+
+        // Always try updating first. Chances are that the branch already exists
+        try
+        {
+            writefln("%s> Updating reference %s", this.repo, refname);
+            reference = con.updateReference(fork_owner,
+                fork_name, refname, commit["sha"].get!string);
+        }
+        catch (HTTPAPIException exc)
+        {
+            writef("%s> Failed to update reference, trying create ... ",
+                this.repo);
+
+            stdout.flush();
+
+            reference = con.createReference(fork_owner,
+                fork_name, refname, commit["sha"].get!string);
+
+            writefln("Ok");
+        }
+
+        auto title =
+            format(PRTitleFormat, this.submod.name, this.submod_new.ver);
+
+        auto content = format(
+"This PR has been automatically created by *neptune-autopr* and
+updates submodule **%s** from version **%s** to version **%s**
+
+You can modify this behavior using your `.neptune.yml` file.
+See the according [documentation](https://github.com/sociomantic-tsunami/neptune/blob/master/doc/neptune-metadata.rst#automatic-update-prs).
```

Comment:
`See the according documentation` sounds weird. It might be better to keep it simple as:
```
You can modify this behavior using your `.neptune.yml` file as described [here](https://github.com/sociomantic-tsunami/neptune/blob/master/doc/neptune-metadata.rst#automatic-update-prs).
```